lxd/storage/drivers/driver_cephfs.go

```diff
@@ -0,0 +1,977 @@
+		newPath: newPath,
+	})
+
+	revertPaths = nil
+	return nil
+}
+
+func (d *cephfs) UpdateVolume(vol Volume, changedConfig map[string]string) error {
+	value, ok := changedConfig["size"]
+	if !ok {
+		return nil
+	}
+
+	size, err := units.ParseByteSizeString(value)
+	if err != nil {
+		return err
+	}
+
+	_, err = shared.RunCommand("setfattr", "-n", "ceph.quota.max_bytes", "-v", fmt.Sprintf("%d", size), GetVolumeMountPath(vol.pool, vol.volType, vol.name))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *cephfs) GetVolumeUsage(volType VolumeType, volName string) (int64, error) {
+	return -1, nil
+}
+
+func (d *cephfs) SetVolumeQuota(volType VolumeType, volName, size string, op *operations.Operation) error {
+	_, err := shared.RunCommand("setfattr", "-n", "ceph.quota.max_bytes", "-v", size, GetVolumeMountPath(d.name, volType, volName))
+	return err
+}
+
+func (d *cephfs) MountVolume(volType VolumeType, volName string, op *operations.Operation) (bool, error) {
+	if volType != VolumeTypeCustom {
+		return false, fmt.Errorf("Volume type not supported")
+	}
+
+	return false, nil
+}
+
+func (d *cephfs) MountVolumeSnapshot(volType VolumeType, VolName, snapshotName string, op *operations.Operation) (bool, error) {
+	if volType != VolumeTypeCustom {
+		return false, fmt.Errorf("Volume type not supported")
+	}
+
+	return false, nil
+}
+
+func (d *cephfs) UnmountVolume(volType VolumeType, volName string, op *operations.Operation) (bool, error) {
+	if volType != VolumeTypeCustom {
+		return false, fmt.Errorf("Volume type not supported")
+	}
+
+	return false, nil
+}
+
+func (d *cephfs) UnmountVolumeSnapshot(volType VolumeType, volName, snapshotName string, op *operations.Operation) (bool, error) {
+	if volType != VolumeTypeCustom {
+		return false, fmt.Errorf("Volume type not supported")
+	}
+
+	return false, nil
+}
+
+func (d *cephfs) CreateVolumeSnapshot(volType VolumeType, volName string, newSnapshotName string, op *operations.Operation) error {
+	if volType != VolumeTypeCustom {
+		return fmt.Errorf("Volume type not supported")
+	}
+
+	// Create the snapshot
+	sourcePath := GetVolumeMountPath(d.name, volType, volName)
+	cephSnapPath := filepath.Join(sourcePath, ".snap", newSnapshotName)
+
+	err := os.Mkdir(cephSnapPath, 0711)
+	if err != nil {
+		return err
+	}
+
+	targetPath := GetVolumeMountPath(d.name, volType, GetSnapshotVolumeName(volName, newSnapshotName))
+
+	err = os.MkdirAll(filepath.Dir(targetPath), 0711)
+	if err != nil {
+		return err
+	}
+
+	err = os.Symlink(cephSnapPath, targetPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *cephfs) DeleteVolumeSnapshot(volType VolumeType, volName string, snapshotName string, op *operations.Operation) error {
+	if volType != VolumeTypeCustom {
+		return fmt.Errorf("Volume type not supported")
+	}
+
+	sourcePath := GetVolumeMountPath(d.name, volType, volName)
+	cephSnapPath := filepath.Join(sourcePath, ".snap", snapshotName)
+
+	err := os.RemoveAll(cephSnapPath)
```

Comment:
Should just be `os.Remove`, it's a magic directory and we shouldn't try to recurse.

Reply:
Fixed